search/search_repository.go

```diff
@@ -307,6 +310,30 @@ func parseMap(queryMap map[string]interface{}, q *Query) {
 	}
 }
 
+func parseOptions(queryMap map[string]interface{}, q *Query) {
+	for key, val := range queryMap {
+		switch val.(type) {
+		case []interface{}:
+			if key == OPTS {
```

Comment:
I think the three lines above can be shortened to just:

```go
for key, val := range queryMap {
  if key==OPTS && ifMap, ok := val.([]interface{}); ok {
    
  }
}
```

Then you can re-use `ifMap` in the next for loop and replace  `for _, v := range val.([]interface{}) {` with `for _, v := range ifMap) {`.

Reply:
Done. 3b7714616807204e908ff9d68fa4d1a072d17a1d